File: unidbg-android/src/main/java/com/vortexdbg/AndroidEmulator.java
package com.vortexdbg;

import com.vortexdbg.arm.ARMEmulator;
import com.vortexdbg.file.linux.AndroidFileIO;
import com.vortexdbg.linux.android.dvm.VM;

import java.io.File;

public interface AndroidEmulator extends ARMEmulator<AndroidFileIO> {

    VM createDalvikVM();

    /**
     * @param apkFile 可为null
     */
    VM createDalvikVM(File apkFile);

    /**
     * jar as apk
     */
    VM createDalvikVM(Class<?> callingClass);

    @SuppressWarnings("unused")
    VM getDalvikVM();

}

File: vortexdbg-android/src/main/kotlin/com/vortexdbg/AndroidEmulator.kt
package com.vortexdbg

import com.vortexdbg.arm.ARMEmulator
import com.vortexdbg.file.linux.AndroidFileIO
import com.vortexdbg.linux.android.dvm.VM

import java.io.File

interface AndroidEmulator : ARMEmulator<AndroidFileIO> {

    fun createDalvikVM(): VM

    /**
     * @param apkFile may be null
     */
    fun createDalvikVM(apkFile: File?): VM

    /** Treats the calling class's code source (a jar) as the apk. */
    fun createDalvikVM(callingClass: Class<*>): VM

    @Suppress("unused")
    fun getDalvikVM(): VM?

}
